File: compare2.js
const _ = require('lodash');
const underscore = require('underscore');

console.time('for loop');
let sum = 0;
for (let i = 0; i < 1e7; i++) {
    sum += i;
}
console.timeEnd('for loop');

console.time('map loop');
const arr = Array.from({ length: 1e7 }, (_, i) => i);
let sumMap = 0;
arr.map((num) => {
    sumMap += num;
});
console.timeEnd('map loop');

// Lodash sum
console.time('Lodash sum');
let sumLodash = _.sum(arr);
console.timeEnd('Lodash sum');

// Underscore sum
console.time('Underscore sum');
let sumUnderscore = underscore.reduce(arr, (sum, num) => sum + num, 0);
console.timeEnd('Underscore sum');

File: compare2.py
import time
import numpy as np
import pandas as pd

# For loop
start_time = time.time()
sum_for = 0
for i in range(int(1e7)):
    sum_for += i
print(f"For loop: {time.time() - start_time} seconds")

# List comprehension with sum
start_time = time.time()
sum_comprehension = sum([i for i in range(int(1e7))])
print(f"List comprehension: {time.time() - start_time} seconds")

# NumPy sum
start_time = time.time()
arr = np.arange(1e7)
sum_numpy = np.sum(arr)
print(f"NumPy sum: {time.time() - start_time} seconds")

# Pandas sum
start_time = time.time()
df = pd.Series(range(int(1e7)))
sum_pandas = df.sum()
print(f"Pandas sum: {time.time() - start_time} seconds")